crates/sdk/src/init.rs

```diff
@@ -0,0 +1,108 @@
+use std::{fs::File, io::Write, path::Path};
+
+use eyre::Result;
+use num_bigint::BigUint;
+use openvm_algebra_circuit::{Fp2Extension, ModularExtension};
+use openvm_ecc_circuit::WeierstrassExtension;
+
+// Reads the vm config and generates a init.rs file that contains a call to moduli_init!,
+// complex_init!, sw_init! with the supported moduli and curves.
+pub fn generate_init_file(
+    manifest_dir: &Path,
+    modular_config: &Option<ModularExtension>,
+    complex_config: &Option<Fp2Extension>,
+    weierstrass_config: &Option<WeierstrassExtension>,
+    init_file_name: Option<&str>, // if None, we use "openvm-init.rs"
+) -> Result<()> {
+    let dest_path = Path::new(manifest_dir).join(init_file_name.unwrap_or("openvm-init.rs"));
+    let mut f = File::create(&dest_path)?;
+
+    writeln!(
+        f,
+        r#"// This file is automatically generated by cargo openvm. Do not rename or edit."#,
+    )?;
+
+    write_moduli_init(&mut f, modular_config)?;
+    write_complex_init(&mut f, modular_config, complex_config)?;
+    write_sw_init(&mut f, weierstrass_config)?;
+    Ok(())
+}
+
+fn write_moduli_init(f: &mut File, modular_config: &Option<ModularExtension>) -> Result<()> {
```

Comment:
i feel like each of these belongs with the respective extension? I guess it'd have to be in the `*-circuit` crate